### ISSUE TITLE
SNOW-626566: Please add a SECURITY.md file and security policy

### ISSUE THREAD

**Issue by setharnold** [snowflakedb/gosnowflake#619]
Hello, please add a `SECURITY.md` file and fill out a security policy on this repository: https://github.com/snowflakedb/gosnowflake/security/policy

Please also consider setting a default security policy: https://docs.github.com/en/code-security/getting-started/securing-your-organization

Thanks

**Comment by sfc-gh-bsterne** [comment]
Thank you for reaching out. You can submit a vulnerability report through HackerOne here:
https://hackerone.com/139c0e4f-5b34-470a-b81e-aa8740c3e66e/embedded_submissions/new

Please select Golang Driver under asset. Thanks again.

I'll follow up separately about adding a SECURITY.md

**Comment by setharnold** [comment]
Thanks @sfc-gh-bsterne -- someone else replied to my oss-security posting with https://www.snowflake.com/product/security-and-trust-center/ -- I *searched* for this, I just never found it. So here's hoping scattering the URL around a little bit will help make it more visible.

**Comment by sfc-gh-dszmolka** [comment]
thank you for your patience here and of course submitting this issue ! i'm syncing with the team to see where this request currently is and what are the next steps. will update once I have any news.

**Comment by sfc-gh-dszmolka** [comment]
possibly addressed by https://github.com/snowflakedb/gosnowflake/pull/1076 - thank you for your contribution @danMateer

**Comment by danMateer** [comment]
No problem @sfc-gh-dszmolka. I mentioned in the PR, too, but if it's possible to expand this a bit into a format that looks more like the one below it can help make sure: 
1. contributors new to this GH feature don't both submit to directly to Snowflake (good) - and - open a PR/Issue (bad).
2. contributors sending potential security problems in have a basic guide right in front of them as a reference reminder of helpful info to include.

If this looks better to you lmk and I can update the diff. What's there now was copied from [snowflake-connector-python](https://github.com/snowflakedb/snowflake-connector-python/tree/main) assuming it was approved language from powers-that-be.

```
## Reporting Security Issues
**Please do not report security vulnerabilities through public GitHub issues.** Instead, please refer to the Snowflake [HackerOne program](https://hackerone.com/snowflake?type=team) for our security policies and for reporting any security vulnerabilities.

For other security related questions and concerns, please contact the Snowflake security team at security@snowflake.com

You should receive a prompt response. If for some reason you do not, please follow up via email to ensure we received your original message.

Please include the requested information listed below (as much as you can provide) to help us better understand the nature and scope of the possible issue:

  * Type of issue (e.g. missing encryption of sensitive data, SQL injection, cross-site scripting, etc.)
  * Full paths of source file(s) related to the manifestation of the issue
  * The location of the affected source code (tag/branch/commit or direct URL)
  * Any special configuration required to reproduce the issue
  * Step-by-step instructions to reproduce the issue
  * Proof-of-concept or exploit code (if possible)
  * Impact of the issue, including how an attacker might exploit the issue

This information will help us expedite response and any appropriate remediation.
 
## Preferred Languages
We prefer all communications to be in English.
```
-----------------
## Reporting Security Issues
**Please do not report security vulnerabilities through public GitHub issues.** Instead, please refer to the Snowflake [HackerOne program](https://hackerone.com/snowflake?type=team) for our security policies and for reporting any security vulnerabilities.

For other security related questions and concerns, please contact the Snowflake security team at security@snowflake.com

You should receive a prompt response. If for some reason you do not, please follow up via email to ensure we received your original message.

Please include the requested information listed below (as much as you can provide) to help us better understand the nature and scope of the possible issue:

  * Type of issue (e.g. missing encryption of sensitive data, SQL injection, cross-site scripting, etc.)
  * Full paths of source file(s) related to the manifestation of the issue
  * The location of the affected source code (tag/branch/commit or direct URL)
  * Any special configuration required to reproduce the issue
  * Step-by-step instructions to reproduce the issue
  * Proof-of-concept or exploit code (if possible)
  * Impact of the issue, including how an attacker might exploit the issue

This information will help us expedite response and any appropriate remediation.
 
## Preferred Languages
We prefer all communications to be in English.

**Comment by sfc-gh-dszmolka** [comment]
thank you very much @danMateer  - this is currently under internal review ; to give a short update on the progress.

**Comment by sfc-gh-dszmolka** [comment]
released with 2024 April release cycle, version 1.10.0
